zerver/lib/github.py

```diff
@@ -61,3 +67,38 @@ def get_latest_github_release_download_link_for_platform(platform: str) -> str:
         if verify_release_download_link(link):
             return link
     return "https://github.com/zulip/zulip-desktop/releases/latest"
+
+
+def get_issue_or_pr_cache_key(owner: str, repo: str, issue_number: str) -> str:
+    github_api_auth_token = settings.GITHUB_API_AUTH_TOKEN
+    if github_api_auth_token is not None:
+        return f"github_issue_or_pr_data_{hashlib.sha256(github_api_auth_token.encode()).hexdigest()[:10]}:{owner}/{repo}/{issue_number}"
+    return f"github_issue_or_pr_data:{owner}/{repo}/{issue_number}"
+
+
+@cache_with_key(get_issue_or_pr_cache_key, timeout=600)
+def get_issue_or_pr_data(owner: str, repo: str, issue_number: str) -> Dict[str, Any]:
+    api_url = f"https://api.github.com/repos/{owner}/{repo}/issues/{issue_number}"
+    try:
+        response = GithubSession().get(api_url)
+        response.raise_for_status()
+        data: Dict[str, Any] = response.json()
+        additional_data = {
+            "result": "success",
+            "platform": "github",
+            "type": "pulls" if "pull_request" in data else "issues",
+            "owner": owner,
+            "repo": repo,
+            "issue_number": issue_number,
+        }
+        data.update(additional_data)
+        return data
+    except requests.RequestException as err:
+        status_code = (
+            err.response.status_code
+            if hasattr(err, "response") and hasattr(err.response, "status_code")
+            else None
+        )
+        if status_code in [401, 403, 429]:
+            logger.warning("Unable to fetch data from GitHub %s", str(err))
```

Comment:
`Unable to fetch data from GitHub 403 Client Error: Forbidden for url: https://...` isn't terribly readable -- we want some separator before the `err`, if only a `:`:
```suggestion
            logger.warning("Unable to fetch data from GitHub: %s", str(err))
```

I also wonder if we should avoid caching this set of results, by raising an exception.  Thoughts, @timabbott?

Reply:
Currently, the code caches this information, but we will update it once we make a decision.